addons/im_livechat/controllers/main.py

```diff
@@ -122,17 +121,17 @@ def get_session(self, channel_id, previous_operator_id=None, chatbot_script_id=N
                 }
                 store.add(chatbot_script)
                 store.add(welcome_steps)
-            operator_partner = operator_info['operator_partner']
             channel_info = {
                 "fetchChannelInfoState": "fetched",
                 "id": channel_id,
                 "isLoaded": True,
                 "livechat_operator_id": Store.One(
-                    operator_partner, self.env["discuss.channel"]._store_livechat_operator_id_fields(),
+                    operator_info['operator_partner'], self.env["discuss.channel"]._store_livechat_operator_id_fields(),
                 ),
                 "scrollUnread": False,
                 "channel_type": "livechat",
                 "chatbot": chatbot_data,
+                **processed_extra_channel_params,
```

Comment:
I think it would be better API if the method separately returned "fake fields" for `channel_info` of non-persistent channel, and actual fields that need to be written on the channel.